File: canPermutePalindrome.java
public class Solution {
    public boolean canPermutePalindrome(String s) {
        
        int []frequencies = new int[1024];
        //int []frequencies = new int[Character.getNumericValue('z') - Character.getNumericValue('a') + 1];
        Map<Character, Integer> map = new HashMap<Character, Integer>();
        
        for(char c : s.toCharArray()) {
            
            frequencies[c]++;
            
            if(map.containsKey(c)) {
                
                int fre = map.get(c);
                map.put(c, fre + 1);
            }
            else {
                
                map.put(c, 1);
            }
        }
        
        int noOfOdds = 0;
        
        for(int index = 0; index < frequencies.length && noOfOdds < 2; index++) {
            
            if(frequencies[index]%2 != 0) {
                
                // noOfOdds++;
            }
        }
        
        for(Map.Entry<Character,Integer> entry : map.entrySet()) {
            
            if(entry.getValue() %2 != 0) {
                
                noOfOdds++;
            }
        }
        
        return noOfOdds < 2;
    }
}

File: canPermutePalindrome.cpp
class Solution {
public:
    bool canPermutePalindrome(string s) {
        
        unordered_map<char, int> map_;
        int noOfOdds = 0;
        
        int index = 0;
        
        while(index < s.size()) {
            
            map_[s[index]]++;
            
            ++index;
        }
        
        ; //unordered_map<char, int>::iterator iter;
        
        decltype (map_.begin()) i = map_.begin();
        
        while(i != map_.end() && noOfOdds < 2) {
            
            
            if(i-> second % 2 != 0) {
                
                ++noOfOdds;
            }
            
            ++i;
        }
        
        
        return noOfOdds < 2;
    }
};